[prometheus] linting fails due to deprecated rbac api version

**Describe the bug**

Running `helm lint` fails with the following error:

```
$ helm lint
==> Linting .
[ERROR] templates/rbac/alertmanager-clusterrole.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRole" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRole"
[ERROR] templates/rbac/alertmanager-clusterrolebinding.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRoleBinding"
[ERROR] templates/rbac/pushgateway-clusterrole.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRole" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRole"
[ERROR] templates/rbac/pushgateway-clusterrolebinding.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRoleBinding"
[ERROR] templates/rbac/server-clusterrole.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRole" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRole"
[ERROR] templates/rbac/server-clusterrolebinding.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRoleBinding"

Error: 1 chart(s) linted, 1 chart(s) failed
```

**Version of Helm and Kubernetes**:

```console
$ helm version
version.BuildInfo{Version:"v3.3.1", GitCommit:"249e5215cde0c3fa72e27eb7a30e8d55c9696144", GitTreeState:"dirty", GoVersion:"go1.15"}
```
Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T21:54:15Z", GoVersion:"go1.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.6-1+64f53401f200a7", GitCommit:"64f53401f200a7a5977955a74ad3f997302774ca", GitTreeState:"clean", BuildDate:"2020-07-15T19:59:02Z", GoVersion:"go1.14.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:

`prometheus`

**Which version of the chart**:

`11.13.1`

**What happened**:

Linting fails

**What you expected to happen**:

Linting to succeed

**How to reproduce it** (as minimally and precisely as possible):

- Clone the repository
- Navigate to directory charts/prometheus
- Run Helm (3+) linting with `helm lint`

**Anything else we need to know**:

Nope

--- a/charts/prometheus/Chart.yaml
+++ b/charts/prometheus/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus
-version: 11.13.1
+version: 11.13.2
 appVersion: 2.20.1
 description: Prometheus is a monitoring system and time series database.
 home: https://prometheus.io/

--- a/charts/prometheus/Chart.yaml
+++ b/charts/prometheus/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus
-version: 11.13.2
+version: 11.14.0
 appVersion: 2.20.1
 description: Prometheus is a monitoring system and time series database.
 home: https://prometheus.io/

--- a/charts/prometheus/templates/_helpers.tpl
+++ b/charts/prometheus/templates/_helpers.tpl
@@ -192,6 +192,16 @@ Return the appropriate apiVersion for podsecuritypolicy.
 {{- print "policy/v1beta1" -}}
 {{- end -}}
 {{- end -}}
+{{/*
+Return the appropriate apiVersion for rbac.
+*/}}
+{{- define "rbac.apiVersion" -}}
+{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1" }}
+{{- print "rbac.authorization.k8s.io/v1" -}}
+{{- else -}}
+{{- print "rbac.authorization.k8s.io/v1beta1" -}}
+{{- end -}}
+{{- end -}}
 
 {{/*
 Create the name of the service account to use for the alertmanager component

--- a/charts/prometheus/templates/rbac/alertmanager-clusterrole.yaml
+++ b/charts/prometheus/templates/rbac/alertmanager-clusterrole.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.alertmanager.enabled .Values.rbac.create .Values.alertmanager.useClusterRole (not .Values.alertmanager.useExistingRole) -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRole
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/alertmanager-clusterrolebinding.yaml
+++ b/charts/prometheus/templates/rbac/alertmanager-clusterrolebinding.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.alertmanager.enabled .Values.rbac.create .Values.alertmanager.useClusterRole -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRoleBinding
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/alertmanager-role.yaml
+++ b/charts/prometheus/templates/rbac/alertmanager-role.yaml
@@ -1,6 +1,6 @@
 {{- if and .Values.alertmanager.enabled .Values.rbac.create (eq .Values.alertmanager.useClusterRole false) (not .Values.alertmanager.useExistingRole) -}}
 {{- range $.Values.alertmanager.namespaces }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: Role
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/alertmanager-rolebinding.yaml
+++ b/charts/prometheus/templates/rbac/alertmanager-rolebinding.yaml
@@ -1,6 +1,6 @@
 {{- if and .Values.alertmanager.enabled .Values.rbac.create (eq .Values.alertmanager.useClusterRole false) -}}
 {{ range $.Values.alertmanager.namespaces }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: RoleBinding
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/node-exporter-role.yaml
+++ b/charts/prometheus/templates/rbac/node-exporter-role.yaml
@@ -1,6 +1,6 @@
 {{- if and .Values.nodeExporter.enabled .Values.rbac.create }}
 {{- if or (default .Values.nodeExporter.podSecurityPolicy.enabled false) (.Values.podSecurityPolicy.enabled) }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: Role
 metadata:
   name: {{ template "prometheus.nodeExporter.fullname" . }}

--- a/charts/prometheus/templates/rbac/node-exporter-rolebinding.yaml
+++ b/charts/prometheus/templates/rbac/node-exporter-rolebinding.yaml
@@ -1,6 +1,6 @@
 {{- if and .Values.nodeExporter.enabled .Values.rbac.create }}
 {{- if .Values.podSecurityPolicy.enabled }}
-apiVersion: rbac.authorization.k8s.io/v1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: RoleBinding
 metadata:
   name: {{ template "prometheus.nodeExporter.fullname" . }}

--- a/charts/prometheus/templates/rbac/pushgateway-clusterrole.yaml
+++ b/charts/prometheus/templates/rbac/pushgateway-clusterrole.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.pushgateway.enabled .Values.rbac.create -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRole
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/pushgateway-clusterrolebinding.yaml
+++ b/charts/prometheus/templates/rbac/pushgateway-clusterrolebinding.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.pushgateway.enabled .Values.rbac.create -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRoleBinding
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/server-clusterrole.yaml
+++ b/charts/prometheus/templates/rbac/server-clusterrole.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.server.enabled .Values.rbac.create (empty .Values.server.useExistingClusterRoleName) -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRole
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/server-clusterrolebinding.yaml
+++ b/charts/prometheus/templates/rbac/server-clusterrolebinding.yaml
@@ -1,5 +1,5 @@
 {{- if and .Values.server.enabled .Values.rbac.create (empty .Values.server.namespaces) (empty .Values.server.useExistingClusterRoleName) -}}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRoleBinding
 metadata:
   labels:

--- a/charts/prometheus/templates/rbac/server-rolebinding.yaml
+++ b/charts/prometheus/templates/rbac/server-rolebinding.yaml
@@ -1,7 +1,7 @@
 {{- if and .Values.server.enabled .Values.rbac.create .Values.server.useExistingClusterRoleName .Values.server.namespaces -}}
 {{ range $.Values.server.namespaces -}}
 ---
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: RoleBinding
 metadata:
   labels:

